core/domain/takeout_service_test.py

```diff
@@ -876,6 +929,121 @@ def test_export_data_for_full_user_trivial_is_correct(self):
         expected_images = []
         self.assertEqual(expected_images, observed_images)
 
+    def test_exports_follow_export_policies(self):
+        """Test to ensure that all fields that should be exported
+        per the export policy are exported, and exported in the proper format.
+        """
+        self.set_up_non_trivial()
+
+        # We set up the feedback_thread_model here so that we can easily
+        # access it when computing the expected data later.
+        feedback_thread_model = feedback_models.GeneralFeedbackThreadModel(
+            entity_type=self.THREAD_ENTITY_TYPE,
+            entity_id=self.THREAD_ENTITY_ID,
+            original_author_id=self.USER_ID_1,
+            status=self.THREAD_STATUS,
+            subject=self.THREAD_SUBJECT,
+            has_suggestion=self.THREAD_HAS_SUGGESTION,
+            summary=self.THREAD_SUMMARY,
+            message_count=self.THREAD_MESSAGE_COUNT
+        )
+        feedback_thread_model.put()
+
+        thread_id = feedback_services.create_thread(
+            self.THREAD_ENTITY_TYPE,
+            self.THREAD_ENTITY_ID,
+            self.USER_ID_1,
+            self.THREAD_SUBJECT,
+            self.MESSAGE_TEXT
+        )
+        feedback_services.create_message(
+            thread_id,
+            self.USER_ID_1,
+            self.THREAD_STATUS,
+            self.THREAD_SUBJECT,
+            self.MESSAGE_TEXT
+        )
+
+        # Retrieve all models for export.
+        all_models = [
+            clazz
+            for clazz in self._get_model_classes()
+            if not clazz.__name__ in self.BASE_CLASSES
+        ]
+
+        # Iterate over models and test export policies.
+        for model in all_models:
+            export_method = model.get_model_association_to_user()
+            export_policy = model.get_export_policy()
+            renamed_export_keys = model.get_field_names_for_takeout()
+            exported_property_names = []
+            props_used_as_takeout_ids = []
```

Comment:
props --> properties

(the name is better, though!)

Reply:
Per discussion -- make it a string variable, and do validation to ensure exactly one field is labelled as such.

property_used_as_key_for_takeout_dict

Reply:
Done